shell/browser/atom_download_manager_delegate.cc

```diff
@@ -90,6 +90,7 @@ void AtomDownloadManagerDelegate::OnDownloadPathGenerated(
     const content::DownloadTargetCallback& callback,
     const base::FilePath& default_path) {
   DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
+  base::ThreadRestrictions::ScopedAllowIO allow_io;
```

Comment:
Which operation in this function required this ? The method is already async, it would be better to move that IO operation to a task runner that supported IO and use the response for further processing rather than scoping IO here.

Reply:
This was the stack trace:
```
[...]
#4 0x55c7fda15428 base::DirectoryExists()
#5 0x55c7faa39243 file_dialog::(anonymous namespace)::FileChooserDialog::FileChooserDialog()
#6 0x55c7faa399a2 file_dialog::ShowSaveDialog()
#7 0x55c7fa9fe197 electron::AtomDownloadManagerDelegate::OnDownloadPathGenerated()
[...]
```

I'm all for moving this to a different thread where IO is allowed, but I didn't want that work to block this merge.